torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1356,6 +1360,33 @@ def _free_full_params_and_use_local_shard(params_to_free):
                         _free_full_params_and_use_local_shard(currently_local_params)
                         self.training_state = TrainingState_.IDLE
 
+    def named_parameters(
+        self,
+        *args,
+        **kwargs,
+    ) -> Iterator[Tuple[str, torch.nn.Parameter]]:
+        """
+        Overrides :meth:`torch.nn.Module.named_parameters()` to intercept
+        parameter names and remove all occurrences of the FSDP-specific
+        flattened parameter prefix when inside the :meth:`summon_full_params`
+        context manager.
+        """
+        # Determine which logic to use based on the context at call time
+        if getattr(self, "training_state", None) != TrainingState_.SUMMON_FULL_PARAMS:
+            for param_name, param in torch.nn.Module.named_parameters(
```

Comment:
Wonder if we can use `super().named_parameters` rather than torch.nn.Module.named_parameters call? If user writes `MyModule` that inherits from `nn.Module` with a custom named_parameters will this work?

Reply:
Great point.